torch/nn/modules/module.py

```diff
@@ -551,6 +551,81 @@ def __repr__(self):
         tmpstr = tmpstr + ')'
         return tmpstr
 
+    def summary(self, input_size):
+        def register_hook(module):
+            def hook(module, input, output):
+                class_name = str(module.__class__).split('.')[-1].split("'")[0]
+                module_idx = len(summary)
+                m_key = '%s-%i' % (class_name, module_idx + 1)
+                summary[m_key] = OrderedDict()
+                summary[m_key]['input_shape'] = list(input[0].size())
+                summary[m_key]['input_shape'][0] = None
+                summary[m_key]['output_shape'] = list(output.size())
+                summary[m_key]['output_shape'][0] = None
+
+                params = 0
+                if hasattr(module, 'weight'):
```

Comment:
This looks a bit fragile to me, because the parameters might be named differently than just `weight` or `bias`.
It is more reliable to use `module.parameters()`, or, if you don't want to walk recursively in the model parameters, to iterate over the [returned values of `module._parameters`](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L437-L439).

Reply:
Thanks, done